### PR TITLE
Feat: add new function to infer git ref

### PR DESCRIPTION
**Pull request by SebastianMorawiec**
# The problem
There was no way to infer git ref using GithubImport and private repos
https://zapatacomputing.atlassian.net/browse/ORQSDK-1063

# This PR's solution
Add new function 

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

File: projects/orquestra-sdk/src/orquestra/sdk/_client/_base/_dsl.py [hunk 1]
```diff
@@ -90,19 +90,63 @@ class GitImportWithAuth:
...
+            "zapatacomputing/my_source_repo",
+            git_ref=sdk.ref_infer(),
+        )
+    )
+        def demo_task():
+            pass
+
+    Usage for other scenarios:
+    def my_fun():
+        sdk.ref_infer("my/path/").infer()
```
Comment (jamesclark-Zapata) [inline projects/orquestra-sdk/src/orquestra/sdk/_client/_base/_dsl.py:139]: **question:** ‏does this need to say `sdk.ref_infer("my/path/").resolve()`?

File: projects/orquestra-sdk/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,8 @@
 
 🔥 *Features*
 
+* New function: `sdk.ref_infer` to infer git refs for Git Imports
```
Comment (jamesclark-Zapata) [inline projects/orquestra-sdk/CHANGELOG.md:9]: **thought:** ‏I'm not sold on the name, it seems to me that `ref` could be anything and not specific it Git.
Having said that, naming things is hard and what we have is better than delaying evan more so we can keep using it and can always rename later, at the cost of a "deprecation cycle".

**Review by jamesclark-Zapata — APPROVED** [review]
lgtm, one question on the docstring.

File: projects/orquestra-sdk/CHANGELOG.md (hunk 2 above)
Comment (alexjuda) [reply projects/orquestra-sdk/CHANGELOG.md]: `sdk.infer_git_ref()`?

**Pull request merged**